File: OpenDentBusiness/TableTypes/InsFilingCode.cs
using System;
using System.Collections;
using OpenDentBusiness.DataAccess;

namespace OpenDentBusiness{
	///<summary>An optional field on insplan and claims.  This lets user customize so that they can track insurance types.</summary>
	[DataObject("insfilingcode")]
	public class InsFilingCode : DataObjectBase{

		[DataField("InsFilingCodeNum",PrimaryKey=true,AutoNumber=true)]
		private long insFilingCodeNum;
		private bool insFilingCodeNumChanged;
		///<summary>Primary key.</summary>
		public long InsFilingCodeNum{
			get { return insFilingCodeNum;}
			set { if(insFilingCodeNum!=value) { insFilingCodeNum=value; MarkDirty(); insFilingCodeNumChanged=true; } }
		}
		public bool InsFilingCodeNumChanged{
			get { return insFilingCodeNumChanged; }
		}

		[DataField("Descript")]
		private string descript;
		private bool descriptChanged;
		///<summary>Description of the insurance filing code.</summary>
		public string Descript {
			get { return descript; }
			set { if(descript!=value) { descript=value; MarkDirty(); descriptChanged=true; } }
		}
		public bool DescriptChanged {
			get { return descriptChanged; }
		}

		[DataField("EclaimCode")]
		private string eclaimCode;
		private bool eclaimCodeChanged;
		///<summary>Code for electronic claim.</summary>
		public string EclaimCode {
			get { return eclaimCode; }
			set { if(eclaimCode!=value) { eclaimCode=value; MarkDirty(); eclaimCodeChanged=true; } }
		}
		public bool EclaimCodeChanged {
			get { return eclaimCodeChanged; }
		}

		[DataField("ItemOrder")]
		private int itemOrder;
		private bool itemOrderChanged;
		///<summary>Display order for this filing code within the UI.  0-indexed.</summary>
		public int ItemOrder {
			get { return itemOrder; }
			set { if(itemOrder!=value) { itemOrder=value; MarkDirty(); itemOrderChanged=true; } }
		}
		public bool ItemOrderChanged {
			get { return itemOrderChanged; }
		}
		
		public InsFilingCode Copy(){
			return (InsFilingCode)Clone();
		}	

	}
}

File: OpenDentalWeb/OpenDental/src/com/opendental/odweb/client/tabletypes/InsFilingCode.java
package com.opendental.odweb.client.tabletypes;

import com.google.gwt.xml.client.Document;
import com.opendental.odweb.client.remoting.Serializing;

/** DO NOT MAKE CHANGES TO THIS FILE.  THEY WILL GET OVERWRITTEN BY THE CRUD. */
public class InsFilingCode {
		/** Primary key. */
		public int InsFilingCodeNum;
		/** Description of the insurance filing code. */
		public String Descript;
		/** Code for electronic claim. */
		public String EclaimCode;
		/** Display order for this filing code within the UI.  0-indexed. */
		public int ItemOrder;

		/** Deep copy of object. */
		public InsFilingCode deepCopy() {
			InsFilingCode insfilingcode=new InsFilingCode();
			insfilingcode.InsFilingCodeNum=this.InsFilingCodeNum;
			insfilingcode.Descript=this.Descript;
			insfilingcode.EclaimCode=this.EclaimCode;
			insfilingcode.ItemOrder=this.ItemOrder;
			return insfilingcode;
		}

		/** Serialize the object into XML. */
		public String serialize() {
			StringBuilder sb=new StringBuilder();
			sb.append("<InsFilingCode>");
			sb.append("<InsFilingCodeNum>").append(InsFilingCodeNum).append("</InsFilingCodeNum>");
			sb.append("<Descript>").append(Serializing.escapeForXml(Descript)).append("</Descript>");
			sb.append("<EclaimCode>").append(Serializing.escapeForXml(EclaimCode)).append("</EclaimCode>");
			sb.append("<ItemOrder>").append(ItemOrder).append("</ItemOrder>");
			sb.append("</InsFilingCode>");
			return sb.toString();
		}

		/** Sets all the variables on this object based on the values in the XML document.  Variables that are not in the XML document will be null or their default values.
		 * @param doc A parsed XML document.  Must be valid XML.  Does not need to contain a node for every variable on this object.
		 * @throws Exception DeserializeFromXml is entirely encased in a try catch and will throw exceptions if anything goes wrong. */
		public void deserialize(Document doc) throws Exception {
			try {
				if(Serializing.getXmlNodeValue(doc,"InsFilingCodeNum")!=null) {
					InsFilingCodeNum=Integer.valueOf(Serializing.getXmlNodeValue(doc,"InsFilingCodeNum"));
				}
				if(Serializing.getXmlNodeValue(doc,"Descript")!=null) {
					Descript=Serializing.getXmlNodeValue(doc,"Descript");
				}
				if(Serializing.getXmlNodeValue(doc,"EclaimCode")!=null) {
					EclaimCode=Serializing.getXmlNodeValue(doc,"EclaimCode");
				}
				if(Serializing.getXmlNodeValue(doc,"ItemOrder")!=null) {
					ItemOrder=Integer.valueOf(Serializing.getXmlNodeValue(doc,"ItemOrder"));
				}
			}
			catch(Exception e) {
				throw e;
			}
		}


}
